stingray/modeling/gpmodeling.py

```diff
@@ -37,10 +40,607 @@
+    ax[0].update(
+        {
+            "xscale": "log",
+            "yscale": "linear",
+            "ylabel": r"$P_{\text{true}} - P_{\text{approx}} $",
+        }
+    )
+
+    ax[0].axvline(f_min, color="black", linestyle="--")
+    ax[0].axvline(f_max, color="black", linestyle="--")
+    ax[1].axvline(f_min, color="black", linestyle="--")
+    ax[1].axvline(f_max, color="black", linestyle="--")
+
+    ax[1].fill_between(f, rat_quantiles[0], rat_quantiles[4], color=colors, alpha=0.25)
+    ax[1].fill_between(f, rat_quantiles[1], rat_quantiles[3], color=colors, alpha=0.5)
+    ax[1].plot(f, rat_quantiles[2], color="black", lw=1)
+    ax[1].update(
+        {
+            "xscale": "log",
+            "yscale": "linear",
+            "xlabel": "Frequency",
+            "ylabel": r"${P_{\text{approx}}}/{P_{\text{true}}}$",
+        }
+    )
+
+    legend_elements = [
+        Line2D([0], [0], color=colors, lw=2, label="Median"),
+        Line2D([0], [0], color="k", lw=1, ls="--", label=r"$f_\mathrm{min}, f_\mathrm{max}$"),
+        Patch(facecolor=colors, edgecolor=colors, alpha=0.25, label="95%"),
+        Patch(facecolor=colors, edgecolor=colors, alpha=0.5, label="68%"),
+    ]
+    ax[1].legend(handles=legend_elements, ncol=3, bbox_to_anchor=(1.0, -0.4))
+    fig.align_ylabels(ax)
+    # fig.tight_layout()
+    return fig, ax
+
+
+def SHO_power_spectrum(f, A, f0):
+    """Power spectrum of a stochastic harmonic oscillator.
+
+    Parameters
+    ----------
+    f : jax.Array
+        Frequency array.
+    A : float
+        Amplitude.
+    f0 : float
+        Position.
+    """
+    P = A / (1 + jnp.power((f / f0), 4))
+
+    return P
+
+
+def _get_coefficients_approximation(
+    kernel_type, kernel_params, f_min, f_max, n_approx_components=20, approximate_with="SHO"
+):
+    """
+    Get the coefficients of the approximation of the power law kernel
+    with a sum of SHO kernels or a sum of DRW+SHO kernels.
+
+    Parameters
+    ----------
+    kernel_type: string
+        The type of kernel to be used for the Gaussian Process
+        Only designed for the following Power spectra ["PowL","DoubPowL]
+    kernel_params: dict
+        Dictionary containing the parameters for the kernel
+        Should contain the parameters for the selected kernel
+    f_min: float
+        The minimum frequency of the approximation grid.
+        Should be the lowest frequency of the power spectrum
+    f_max: float
+        The maximum frequency of the approximation grid.
+        Should be the highest frequency of the power spectrum
+    n_approx_components: int
+        The number of components to use to approximate the power law
+    approximate_with: string
+        The type of kernel to use to approximate the power law power spectra
+    """
+    # grid of frequencies for the approximation
+    spectral_points = jnp.geomspace(f_min, f_max, n_approx_components)
+    # build the matrix of the approximation
+    if approximate_with == "SHO":
```

Comment:
Ah, so currently the only approximation is "SHO"? That should probably be documented somewhere. :)